src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -85,6 +86,10 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string paramete
 
             try
             {
+                var configuration = LoadConfiguration(configurations);
+                if (configuration != null)
+                    jsonRuleEngine.FilterRules(configuration);
```

Comment:
I think the rule filtering should be done at an earlier stage in the process. At this point, it would be filtering the rules at each evaluation of a template. It should likely instead happen in the TemplateAnalyzer.Create() or LoadRules().

Reply:
Create is called on line 31 in CommandLineParser.cs

Reply:
Create would be hard, since it's called before the command line is parsed.  But calling LoadRules before analysis would work instead.

Reply:
+1 for `Create()` = hard and that is true that it is doing it at each evaluation. Johnathon, what do you mean by "But calling LoadRules before analysis would work instead"?
Would calling it in CommandLineParse.cs in `SetupCommandLineAPI()` before the first call to `AnalyzeTemplate` and (re)moving all the `configurationsFilePath` references/actions?

Reply:
Oops, I meant `FilterRules` before analysis, not LoadRules.  Call if from the CLI in `AnalyzeDirectory` before analyzing any templates.  (You'd also need to make sure it's only done in `AnalyzeTemplate` if it isn't done for the directory already.)

Also opened #207 to cleanup growing commonality between template and directory analysis.

Reply:
Sorry, let me clarify again after regaining the context - likely what should happen is some new function like `LoadConfiguration` in Analyzer.Core should be invoked before any analysis.  That function would in turn call `FilterRules` on the rule engine.  After this, the templates are analyzed.